### DISCUSSION TITLE
Columns in the database that holds JSON data.

### DISCUSSION THREAD

**Discussion by pyberg** [joomla/joomla-cms#43127] [2. Cold Features]
In Mysql 8 there where introduced a new column type of JSON, which  let the database control the integrity of the JSON structure. From Joomla 5.0, the minimum requirement is MYSQ 8, but JSON is still being stored in varchar defined columns. Isn't it time to use this feature? Remember when an update of J1.5 caused loads of issue with invalid JSON structured data.

**Comment by richard67** [comment]
In MariaDB, which is also supported by Joomla, JSON data columns are handled differently, see https://mariadb.com/kb/en/json-data-type/ . SO as long as we don't have separate MySQL and MariaDB drivers, we are limited to what is supported in the same, compatible way on both kinds of database with the same database driver..
